pkg/receive/handler.go

```diff
@@ -0,0 +1,203 @@
+package receive
+
+import (
+	"context"
+	"fmt"
+	"io/ioutil"
+	stdlog "log"
+	"net"
+	"net/http"
+	"sync/atomic"
+
+	"github.com/go-kit/kit/log"
+	"github.com/go-kit/kit/log/level"
+	"github.com/gogo/protobuf/proto"
+	"github.com/golang/snappy"
+	"github.com/improbable-eng/thanos/pkg/store/prompb"
+	conntrack "github.com/mwitkow/go-conntrack"
+	"github.com/opentracing-contrib/go-stdlib/nethttp"
+	opentracing "github.com/opentracing/opentracing-go"
+	"github.com/prometheus/client_golang/prometheus"
+	"github.com/prometheus/client_golang/prometheus/promhttp"
+	"github.com/prometheus/common/route"
+	promtsdb "github.com/prometheus/prometheus/storage/tsdb"
+	"github.com/prometheus/tsdb"
+)
+
+var (
+	requestDuration = prometheus.NewHistogramVec(
+		prometheus.HistogramOpts{
+			Name:    "prometheus_http_request_duration_seconds",
```

Comment:
We might want `thanos` prefix for those metrics? Or do we want to keep it similar to get advantage of common dashboards?

Reply:
Sorry that was a copy paste mistake. Yes lets make it `thanos_http_...`.